src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Interaction.Windows.vb

```diff
@@ -0,0 +1,94 @@
+            Dim Title As String
+
+            'Get the Assembly name of the calling assembly
+            'Assembly.GetName requires PathDiscovery permission so we try this first
+            'and if it throws we catch the security exception and parse the name
+            'from the full assembly name
+            Try
+                Title = CallingAssembly.GetName().Name
+            Catch ex As SecurityException
+                Dim FullName As String = CallingAssembly.FullName
+                'Find the text up to the first comma. Note, this fails if the assembly has a comma in its name
+                Dim FirstCommaLocation As Integer = FullName.IndexOf(","c)
+                If FirstCommaLocation >= 0 Then
+                    Title = FullName.Substring(0, FirstCommaLocation)
+                Else
+                    'The name is not in the format we're expecting so return an empty string
+                    Title = ""
+                End If
+            End Try
+
+            Return Title
+        End Function
+
+        Public Function MsgBox(ByVal Prompt As Object, Optional ByVal Buttons As MsgBoxStyle = MsgBoxStyle.OkOnly, Optional ByVal Title As Object = Nothing) As MsgBoxResult
+            Dim sPrompt As String = Nothing
+            Dim sTitle As String
+            'Dim vbhost As CompilerServices.IVbHost
+            'Dim ParentWindow As System.Windows.Forms.IWin32Window = Nothing
+            ' A read Parent window is need to make this monction modal
+            Dim ParentWindow As Integer = Nothing
+
+            'vbhost = CompilerServices.HostServices.VBHost
+            'If Not vbhost Is Nothing Then
+            '    ParentWindow = vbhost.GetParentWindow()
+            'End If
+
+            'Only allow legal button combinations to be set, one choice from each group
+            'These bit constants are defined in System.Windows.Forms.MessageBox
+            'Low-order 4 bits (0x000f), legal values: 0, 1, 2, 3, 4, 5
+            '     next 4 bits (0x00f0), legal values: 0, &H10, &H20, &H30, &H40
+            '     next 4 bits (0x0f00), legal values: 0, &H100, &H200
+            If ((Buttons And &HFI) > MsgBoxStyle.RetryCancel) OrElse
+                ((Buttons And &HF0I) > MsgBoxStyle.Information) OrElse
+                ((Buttons And &HF00I) > MsgBoxStyle.DefaultButton3) Then
+                Buttons = MsgBoxStyle.OkOnly
+            End If
+
+            Try
+                If Not Prompt Is Nothing Then
+                    sPrompt = DirectCast(Conversions.ChangeType(Prompt, GetType(String)), String)
+                End If
+            Catch ex As StackOverflowException
+                Throw ex
+            Catch ex As OutOfMemoryException
+                Throw ex
+            Catch ex As System.Threading.ThreadAbortException
+                Throw ex
+            Catch
+                Throw New ArgumentException(SR.GetResourceString(SR.Argument_InvalidValueType2, "Prompt"), "String")
```

Comment:
>"String" [](start = 107, length = 8)

Should `"String"` be an argument to `GetResourceString()` instead? Please add a test for this case.

Reply:
This was for debugging, I will delete

Reply:
`Argument_InvalidValueType2` requires two arguments. Please re-add `"String"`, but as an argument to `GetResourceString`, and please add a test for this exception.

---
In reply to: [227499060](https://github.com/dotnet/corefx/pull/32668#discussion_r227499060) [](ancestors = 227499060)